File: Arrays _101/414. Third Maximum Number/third_max.py
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        max1, max2, max3 = float("-inf"), float("-inf"), float("-inf")

        for num in nums:
            if num > max1:
                max3 = max2
                max2 = max1
                max1 = num
            elif num < max1 and num > max2:
                max3 = max2
                max2 = num
            elif num < max2 and num > max3:
                max3 = num

        if max3 != float("-inf"):
            return max3
        else:
            return max1

File: Arrays  101/414. Third Maximum Number/third_max.cpp
class Solution
{
public:
    int thirdMax(vector<int> &nums)
    {
        long long max1 = LLONG_MIN, max2 = LLONG_MIN, max3 = LLONG_MIN;

        for (auto num : nums)
        {
            if (num > max1)
            {
                max3 = max2;
                max2 = max1;
                max1 = num;
            }
            else if (num < max1 && num > max2)
            {
                max3 = max2;
                max2 = num;
            }
            else if (num < max2 && num > max3)
            {
                max3 = num;
            }
        }
        return max3 == LLONG_MIN ? max1 : max3;
    }
};